play/src/front/Phaser/Game/MapEditor/Tools/EntityEditorTool.ts

```diff
@@ -0,0 +1,375 @@
+
+    private pointerMoveEventHandler!: (pointer: Phaser.Input.Pointer) => void;
+    private pointerDownEventHandler!: (pointer: Phaser.Input.Pointer) => void;
+
+    constructor(mapEditorModeManager: MapEditorModeManager) {
+        super();
+        this.mapEditorModeManager = mapEditorModeManager;
+        this.scene = this.mapEditorModeManager.getScene();
+
+        this.shiftKey = this.scene.input.keyboard.addKey("SHIFT");
+
+        this.entitiesManager = this.scene.getGameMapFrontWrapper().getEntitiesManager();
+        this.gameMapEntities = this.scene.getGameMap().getGameMapEntities();
+
+        this.entityPrefab = undefined;
+        this.entityPrefabPreview = undefined;
+
+        this.subscribeToStores();
+        this.bindEntitiesManagerEventHandlers();
+    }
+
+    public update(time: number, dt: number): void {}
+    public clear(): void {
+        mapEntityEditorModeStore.set(MapEntityEditorMode.AddMode);
+        this.entitiesManager.clearAllEntitiesTint();
+        this.cleanPreview();
+        this.unbindEventHandlers();
+    }
+    public activate(): void {
+        this.bindEventHandlers();
+    }
+    public destroy(): void {
+        this.cleanPreview();
+        this.unbindEventHandlers();
+        this.mapEditorSelectedEntityPrefabStoreUnsubscriber();
+        this.mapEntityEditorModeStoreUnsubscriber();
+    }
+    public subscribeToGameMapFrontWrapperEvents(gameMapFrontWrapper: GameMapFrontWrapper): void {
+        console.log("EntityEditorTool subscribeToGameMapFrontWrapperEvents");
+    }
+    public handleKeyDownEvent(event: KeyboardEvent): void {
+        console.log("EntityEditorTool handleKeyDownEvent");
+    }
+    /**
+     * Perform actions needed to see the changes instantly
+     */
+    public handleCommandExecution(commandConfig: CommandConfig): void {
+        switch (commandConfig.type) {
+            case "UpdateEntityCommand": {
+                this.handleEntityUpdate(commandConfig.dataToModify);
+                break;
+            }
+            case "CreateEntityCommand": {
+                this.handleEntityCreation(commandConfig.entityData);
+                break;
+            }
+            case "DeleteEntityCommand": {
+                this.handleEntityDeletion(commandConfig.id);
+                break;
+            }
+            default: {
+                break;
+            }
+        }
+    }
+    /**
+     * React on commands coming from the outside
+     */
+    public handleIncomingCommandMessage(editMapCommandMessage: EditMapCommandMessage): void {
+        const commandId = editMapCommandMessage.id;
+        switch (editMapCommandMessage.editMapMessage?.message?.$case) {
+            case "createEntityMessage": {
+                const data = editMapCommandMessage.editMapMessage?.message.createEntityMessage;
+                const entityPrefab = mapEntitiesPrefabsStore.getEntityPrefab(data.collecionName, data.prefabId);
+
```

Comment:
In the future, I'm wondering what will happen if a collection is not hosted on the server.

What happens if someone imports a map referencing a collection declared somewhere else? Should we store the URL of the collection (next to its name) in the JSON describing entities?